File: tests/fixtures/code-evidence/java/OrderApplicationService.java
package com.acme.orders.application;

import java.time.Clock;
import java.time.Instant;
import java.util.List;
import java.util.Optional;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

/** Coordinates the order aggregate and its transactional ports. */
@Service
public final class OrderApplicationService {
    private final OrderRepository repository;
    private final DomainEventPublisher publisher;
    private final Clock clock;

    @Value("${orders.region:eu-west}")
    private String region;

    public OrderApplicationService(
        OrderRepository repository,
        DomainEventPublisher publisher,
        Clock clock
    ) {
        this.repository = repository;
        this.publisher = publisher;
        this.clock = clock;
    }

    /** Place an order exactly once for a client request identifier. */
    @Transactional
    public Order place(PlaceOrder command) {
        validate(command);
        Optional<Order> existing = repository.findByRequestId(command.requestId());
        if (existing.isPresent()) {
            return existing.get();
        }

        Instant now = clock.instant();
        Order order = Order.place(
            command.requestId(),
            command.customerId(),
            List.copyOf(command.lines()),
            region,
            now
        );
        repository.save(order);
        publish(order);
        return order;
    }

    public Optional<Order> find(String id) {
        if (id == null || id.isBlank()) {
            return Optional.empty();
        }
        return repository.findById(id);
    }

    private void validate(PlaceOrder command) {
        if (command == null) {
            throw new IllegalArgumentException("command is required");
        }
        if (command.lines().isEmpty()) {
            throw new IllegalArgumentException("at least one line is required");
        }
    }

    private void publish(Order order) {
        for (DomainEvent event : order.releaseEvents()) {
            publisher.publish(event);
        }
    }
}

enum OrderState {
    DRAFT,
    PLACED,
    CANCELLED;

    public boolean terminal() {
        return this == CANCELLED;
    }
}

File: tests/fixtures/code-evidence/php/OrderApplicationService.php
<!doctype html>
<html lang="en">
<body data-controller="orders">
    <p>The application service is loaded by the runtime container.</p>
</body>
</html>
<?php
declare(strict_types=1);

namespace App\Orders\Application;

use App\Orders\Domain\Order;
use App\Orders\Domain\PlaceOrder;
use App\Orders\Port\EventPublisher;
use App\Orders\Port\OrderRepository;
use DateTimeImmutable;
use RuntimeException;

/** Coordinates the transactional order-placement use case. */
final readonly class OrderApplicationService
{
    public function __construct(
        private OrderRepository $orders,
        private EventPublisher $events,
    ) {
    }

    /** Persist one order and publish all resulting domain events. */
    public function place(PlaceOrder $command): Order
    {
        $existing = $this->orders->findByRequestId($command->requestId);
        if ($existing !== null) {
            return $existing;
        }

        $order = Order::place(
            $command->requestId,
            $command->customerId,
            $command->lines,
            new DateTimeImmutable(),
        );

        $this->orders->save($order);
        foreach ($order->releaseEvents() as $event) {
            $this->events->publish($event);
        }

        return $order;
    }

    public function describe(Order $order): string
    {
        $region = config('orders.region');
        $message = <<<TEXT
Order {$order->id()} is handled in {$region}.
No function fakeHeredocSymbol() exists here.
TEXT;
        return $message;
    }

    private function requireQueue(): string
    {
        $queue = getenv('ORDERS_QUEUE');
        if ($queue === false || $queue === '') {
            throw new RuntimeException('ORDERS_QUEUE is required');
        }
        return $queue;
    }
}

function normalize_order_reference(string $reference): string
{
    return strtolower(trim($reference));
}
?>
<script>
function fakeTemplateFunction() {
    return "not PHP";
}
</script>
